homeassistant/components/plugwise/config_flow.py

```diff
@@ -166,30 +198,60 @@ async def async_step_user(
                 user_input[CONF_PORT] = self.discovery_info.port
                 user_input[CONF_USERNAME] = self._username
 
-            try:
-                api = await validate_input(self.hass, user_input)
-            except ConnectionFailedError:
-                errors[CONF_BASE] = "cannot_connect"
-            except InvalidAuthentication:
-                errors[CONF_BASE] = "invalid_auth"
-            except InvalidSetupError:
-                errors[CONF_BASE] = "invalid_setup"
-            except (InvalidXMLError, ResponseError):
-                errors[CONF_BASE] = "response_error"
-            except UnsupportedDeviceError:
-                errors[CONF_BASE] = "unsupported"
-            except Exception:  # noqa: BLE001
-                errors[CONF_BASE] = "unknown"
-            else:
+            con_res = await verify_connection(self.hass, user_input)
+            if not isinstance(con_res, dict):
```

Comment:
```suggestion
           if isinstance(con_res, Smile):
```

Reply:
Yeah, bot the 'not's make sense, but than we're in a new rabbit hole called correctly Mocking - as it stands, this won't work since all are calls are MagicMock, not Smile :(
I'm open to (a) learning (curve) but up to now it takes considerable effort to the conftest + adjusting other tests (or introducing one separately for config_flow).
Not trying the easy way out, but it's a lot less code to have just the inverse present (though as said, happy to learn a decent alternative).

Reply:
@abmantis we tried this but `Smile` is not a type `isinstance()` recognizes, therefore using `not` and `dict` is the only available option.

Reply:
You are right, mocking gets too cumbersome :/ In that case I think the original implementation (with a `tuple`) would be better here. 
Are you ok with reverting back to the `tuple`? Sorry for the back-and-forward.

Reply:
I'll rework instead of reverse ...